internal/rsl/rsl.go

```diff
@@ -513,6 +514,64 @@ func GetFirstReferenceEntryForRef(repo *git.Repository, targetRef string) (*Refe
 	return firstEntry, annotations, nil
 }
 
+// SkipAllInvalidReferenceEntriesForRef identifies invalid RSL reference entries.
+// Each invalid entry points to a target that is not reachable for the current
+// target of the same reference, indicating that history has been rewritten via a
+// rebase for the reference. After the invalid entries are identified, an annotation
+// entry is created that marks all of these entries as to be skipped.
+func SkipAllInvalidReferenceEntriesForRef(repo *git.Repository, targetRef string, signCommit bool) error {
+	slog.Debug("Checking if RSL entries point to non-existent commits...")
+
+	latestEntry, _, err := GetLatestReferenceEntryForRef(repo, targetRef)
+	iterator := Entry(latestEntry)
+	if err != nil {
+		return err
+	}
+
+	entriesToSkip := []plumbing.Hash{}
+
+	for {
+		if entry, ok := iterator.(*ReferenceEntry); ok {
+			latestEntryCommit, err := gitinterface.GetCommit(repo, latestEntry.TargetID)
+
+			var isAncestor bool
+
+			switch {
+			case errors.Is(err, plumbing.ErrObjectNotFound):
+				// the current commit cannot be reached from the latest entry, since it does not exist
+				isAncestor = false
+			case err == nil:
+				isAncestor, err = gitinterface.KnowsCommit(repo, entry.TargetID, latestEntryCommit)
+				if err != nil {
+					return err
+				}
+			default:
+				return err
+			}
+			if !isAncestor {
+				slog.Debug(fmt.Sprintf("Found RSL entry '%s' pointing to a commit, '%s', that does not exist in the target ref.", entry.ID, entry.TargetID))
+				entriesToSkip = append(entriesToSkip, entry.ID)
+			} else {
+				slog.Debug(fmt.Sprintf("Found RSL entry '%s' pointing to a commit, '%s', that does exist in the target ref. No more commits to skip.", entry.ID, entry.TargetID))
```

Comment:
/s/that does exist/that exists

maybe you could also log the target ref?

Reply:
I have added the target ref in front of both slog.Debug statements.